control-api/client.go

```diff
@@ -112,21 +112,25 @@ func (api *Client) EnterLameDuck(nodeId string) (*LameDuckResponse, error) {
 	return &response, nil
 }
 
-// Functions the same as ListNodes (ping), except that only nodes currently running
-// the workload in question will respond. Useful to answer the questions, "is this workload running?"
-// and if so, on which node?
-func (api *Client) ListNodesWithWorkload(workloadId string) ([]PingResponse, error) {
+// This is a filtered node ping that returns only matching workloads.
+// A workloadId of "" will not filter by workload, and only
```

Comment:
`nil` (e.g., `workloadID *string`) would probably be better to read and document.

Reply:
I disagree. I've lost all my patience with the `*string` pattern. Empty strings are just as easy to read here and half as much of a PITA